### PR TITLE
Move fixed entries to the "resolved" directory

### PR DESCRIPTION
**Pull request by ksy36**

### CODE REVIEW

**Comment by jgraham** [comment]
Can we do `resolved/<version>` so that we know the releases in which things shipped?

**Comment by jgraham** [comment]
Also we want to exclude these from scoring, but we could do that as a followup if you like.

File: tools/kbcheck/src/data.rs [hunk 1]
```diff
@@ -29,13 +29,21 @@ pub fn read_json(path: &Path) -> Result<serde_json::Value, DataError> {
...
+
 /// Iterator over DirEntry values corresponding to knowledge base data
 /// files
 pub fn iter_data_files(root_path: &Path) -> impl Iterator<Item = Result<DirEntry, DataError>> {
     let data_path = root_path.to_owned().join("data");
 
     WalkDir::new(&data_path)
         .into_iter()
+        .filter_entry(|e| !is_resolved(e))
         .filter(|maybe_entry| {
```
Comment (ksy36) [inline tools/kbcheck/src/data.rs:47]: not sure if using an additional filter here is the best way (sorry, not familiar with rust :/ )

Comment (denschub) [reply tools/kbcheck/src/data.rs:47]: This now would also exclude resolved entries from validation, not just from scoring. I think it's a not-bad idea to keep validating the resolved items, but not score them, so we'd probably need a `ignore_resolved` argument here or something. @jgraham, wdyt?

File: tools/kbcheck/src/data.rs [hunk 2]
```diff
@@ -29,13 +29,21 @@ pub fn read_json(path: &Path) -> Result<serde_json::Value, DataError> {
     Ok(serde_json::from_reader(reader)?)
 }
 
+fn is_resolved(entry: &DirEntry) -> bool {
+    entry.path()
+         .to_str()
+         .map(|s| s == "data/resolved")
```
Comment (denschub) [inline tools/kbcheck/src/data.rs:35]: You probably want to check if [`s.starts_with("data/resolved")`](https://doc.rust-lang.org/nightly/std/string/struct.String.html#method.starts_with) to handle subdirectories as well.

File: tools/kbcheck/src/data.rs (hunk 1 above)
Comment (jgraham) [reply tools/kbcheck/src/data.rs:47]: Yes, I was planning to add a `include_resolved: bool` parameter. I'll fix this up quickly so we can exclude resolved bugs from the report.

**Review by denschub — APPROVED** [review]

File: tools/kbcheck/src/data.rs (hunk 1 above)
Comment (ksy36) [reply tools/kbcheck/src/data.rs:43]: Thanks!!

**Pull request merged**